mrp_multi_level/wizards/mrp_inventory_procure.py

```diff
@@ -134,18 +198,168 @@ class MrpInventoryProcureItem(models.TransientModel):
+        help="Vendor selected from the product's vendor list (seller info).",
+    )
+    currency_id = fields.Many2one(
+        comodel_name="res.currency",
+        string="Currency",
+        help="Currency that will be used for a purchase procurement.",
+    )
+    original_qty = fields.Float(
+        string="Original Planned Qty",
+        help="Original quantity from planned order, used as reference",
+        readonly=True,
+    )
+    source_context = fields.Selection(
+        selection=[
+            ("inventory", "From Inventory"),
+            ("planned_order", "From Planned Order"),
+        ],
+        default="inventory",
+        readonly=True,
+    )
+    mrp_action = fields.Selection(
+        selection=[
+            ("manufacture", "Manufacturing Order"),
+            ("buy", "Purchase Order"),
+            ("pull", "Pull From"),
+            ("push", "Push To"),
+            ("pull_push", "Pull & Push"),
+            ("none", "None"),
+        ],
+        string="Procurement Method",
+        help="Method to use for procurement. Can be modified before executing.",
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super().create(vals_list)
+        for rec in records:
+            if rec.mrp_action == "buy" and rec.product_id and rec.warehouse_id:
+                rec._onchange_purchase_defaults()
+        return records
+
+    @api.onchange(
+        "product_id", "warehouse_id", "qty", "uom_id", "date_planned", "mrp_action"
+    )
+    def _onchange_purchase_defaults(self):
+        """Pre-fill vendor/currency for buy procurements so the user can see them."""
+        for rec in self:
+            if rec.mrp_action != "buy" or not rec.product_id or not rec.warehouse_id:
+                rec.supplier_id = False
+                rec.currency_id = False
+                continue
+
+            company = rec.warehouse_id.company_id or rec.env.company
+            qty = rec.qty or 0.0
+            uom = rec.uom_id or rec.product_id.uom_id
+
+            seller = rec.product_id._select_seller(
+                quantity=qty,
+                date=rec.date_planned,
+                uom_id=uom,
+                partner_id=False,
+            )
+            if not seller:
+                seller = rec.product_id.seller_ids.filtered(
+                    lambda s: (not s.company_id or s.company_id == company)
+                ).sorted(lambda s: (s.sequence, s.min_qty, s.price))[:1]
+
+            rec.supplier_id = seller.partner_id if seller else False
+            rec.currency_id = (
+                seller.currency_id
+                if (seller and seller.currency_id)
+                else company.currency_id
+            )
+
+    def _get_procurement_rule(self):
+        self.ensure_one()
+
+        if not self.mrp_action or self.mrp_action == "none":
+            return False
+
+        location = self.location_id
+        warehouse = self.warehouse_id
+
+        domain = [
```

Comment:
```suggestion
      company = self.warehouse_id.company_id or self.env.company
      domain = [
          ("action", "=", self.mrp_action),
          ("location_dest_id", "=", location.id),
          ("company_id", "in", [company.id, False]),
      ]
      rule = self.env["stock.rule"].search(domain, order="sequence", limit=1)
      
      if not rule:
          domain2 = [
              ("action", "=", self.mrp_action),
              ("warehouse_id", "=", self.warehouse_id.id),
              ("company_id", "in", [company.id, False]),
          ]
          rule = self.env["stock.rule"].search(domain2, order="sequence", limit=1)
```

Si hay varias reglas con el mismo action (ej: dos reglas "buy"), limit=1 sin order puede devolver cualquiera. Añadimos order="sequence" para que devuelva la de mayor prioridad.

 En entorno multi-company podría devolver una regla de otra compañía. Añadimos filtro por company_id.